src/hotkeys_provider.rs

```diff
@@ -125,72 +116,69 @@ pub fn HotkeysProvider(
         });
     }
 
-    #[cfg(any(feature = "hydrate", feature = "csr"))]
-    div()
-        .on_mount(move |_| {
-            let blur_listener = Closure::wrap(Box::new(move || {
-                if cfg!(feature = "debug") {
-                    logging::log!("Window lost focus");
-                }
-                pressed_keys.set(HashMap::new());
-            }) as Box<dyn Fn()>);
-
-            let keydown_listener = Closure::wrap(Box::new(move |event: KeyboardEvent| {
-                pressed_keys.update(|keys| {
-                    keys.insert(event.key().to_lowercase(), event);
-                });
-            }) as Box<dyn Fn(_)>);
-            let keyup_listener = Closure::wrap(Box::new(move |event: KeyboardEvent| {
-                pressed_keys.update(|keys| {
-                    keys.remove(&event.key().to_lowercase());
-                });
-            }) as Box<dyn Fn(_)>);
+    node_ref.on_load(move |_| {
```

Comment:
so for `NodeRef`, the closest function to `on_mount` was `on_load`: https://docs.rs/leptos_dom/latest/leptos_dom/struct.NodeRef.html#method.on_load

I'm just a bit concerned that this might possibly rerender? @mondeja 

@maccesch what do you think? Is this reasonable? Instead of providing a component, I wanted to pack everything in a `provide_hotkeys_context()`. Leptos-use does a great job with this

Reply:
If you look at the documentation of leptos-use, it always gives the programmer the responsibility of creating the node refs with `create_node_ref` and passing it to functions, for example see [`use_element_size`](https://leptos-use.rs/elements/use_element_size.html), but these node refs can be signals also:

```rust
pub fn use_element_size<El, T>(target: El) -> UseElementSizeReturn
where
    El: Into<ElementMaybeSignal<T, web_sys::Element>> + Clone,
    T: Into<web_sys::Element> + Clone + 'static,
{
    ...
}
```

Note the `Into<ElementMaybeSignal<Into<web_sys::Element> + Clone + 'static, web_sys::Element>> + Clone`. Can be a signal to a something that can be converted to a `web_sys::Element` or an `Element` itself. In something like that I was thinking when I said that the node refs could be passed as argument instead of creating a `<div>` element. This pattern is fully reactive, so you could ideally change the ref dinamically.

I'm pretty sure that `on_load` will not rerender as it takes a `FnOnce`, see https://docs.rs/leptos_dom/latest/src/leptos_dom/node_ref.rs.html#168